### DISCUSSION TITLE
Extract text from annotation in certain area/rectangle?

### DISCUSSION THREAD

**Discussion by jiezhoujenny** [pymupdf/PyMuPDF#1539] [Looking for help] [answered]
Is it possible to get annotation text from certain area? like the area in red rectangle shows in attached picture?
Or is there any way to get the annotation position?
Thank you in advance
![image](https://user-images.githubusercontent.com/17275074/149638271-acd9bd9e-4e6c-4739-a194-04f564ae6369.png)

**Comment by JorjMcKie** [comment, marked as answer]
Yes, that should be possible.
1. Each annot has a rect wihich denotes its position on the page. Use it to subselect the annots that are in the red rectangle.
2. For the selected annots, use their inbuilt `get_text()` method - presumably they are FreeText annots in your case.

**Reply by jiezhoujenny** [reply]
@JorjMcKie thank you for instant help, very appreciate!
For viewer who may also interested. here is the code:
1
`for annot in page.annots():
    print(annot.rect)`
---this will get all the annotation rectangle positions
2
`rect=(0.0, 170.0, 612.0, 792.0)
print(page.get_textbox(rect))`
---- this may print all the text on this page, including the text in annotation
